client/src/controllers/InitController.ts

```diff
@@ -14,20 +14,41 @@ import { debounce } from '../utils/debounce';
  * <div class="keep-me hide-me" data-controller="w-init" data-w-init-event-value="custom:event other-custom:event">
  *   When the DOM is ready, two additional custom events will be dispatched; `custom:event` and `other-custom:event`.
  * </div>
+ *
+ * @example - Detail dispatching
+ * <div class="keep-me" data-controller="w-init" data-w-init-detail-value='{"status": "success", "message": "Detail is ready"}'>
+ *  When the DOM is ready, the detail with value of a JSON object: {"status": "success", "message": "Detail is ready"} will be dispatched.
+ * </div>
  */
 export class InitController extends Controller<HTMLElement> {
   static classes = ['ready', 'remove'];
 
   static values = {
     delay: { default: -1, type: Number },
+    detail: { default: {}, type: Object },
     event: { default: '', type: String },
   };
 
-  declare readonly readyClasses: string[];
-  declare readonly removeClasses: string[];
+  /**
+   * The delay before applying ready classes and dispatching events.
+   */
+  declare delayValue: number;
 
+  /**
+   * The detail value to be dispatched with events when the element is ready.
+   */
+  declare detailValue: Record<string, unknown>;
+
+  /**
+   * The custom events to be dispatched when the element is ready.
+   */
   declare eventValue: string;
-  declare delayValue: number;
+
+  /**
+   * The classes to be added and removed when the element is ready.
+   */
+  declare readonly readyClasses: string[];
+  declare readonly removeClasses: string[];
```

Comment:
Thank you for updating this, just a small tweak, let's keep the JSDoc on single lines and also make sure the two classes items have their own JSDoc. Here's a suggestion.


```ts
  /** The delay before applying ready classes and dispatching events. If below zero, no delay will be used. */
  declare delayValue: number;
  /** The detail value to be dispatched with events when the element is ready. */
  declare detailValue: Record<string, unknown>;
  /** Event names to use when dispatching additional custom events when ready. */
  declare eventValue: string;

  /** The classes to be added when the element is ready. */
  declare readonly readyClasses: string[];
  /** The classes to be removed when the element is ready. */
  declare readonly removeClasses: string[];
```